types/tampermonkey/index.d.ts

```diff
@@ -239,6 +224,123 @@ declare namespace Tampermonkey {
     }
 
     type NotificationDetails = TextNotification | HighlightNotification;
+
+    interface GM_info {
+        downloadMode: 'native' | 'browser' | 'disabled';
+        isFirstPartyIsolation: boolean | undefined;
+        isIncognito: boolean;
+
+        /**
+         * `.. | null` means if it was not explicitely set in the metadata
+         * block it is null
+         */
+        script: {
+            antifeatures: Record<string, Record<string, string>>;
+            author: string | null;
+
+            /**
+             * Idk what this is, nothing I did changed this from any empty array
+             * and it's not documented anywhere
+             */
+            blockers: any[];
+
+            copyright: string | null;
+            description: string | null;
+            description_i18n: Record<string, string>;
+            downloadURL: string | null;
+            evilness: number;
+            excludes: string[];
+            grant: string[];
+            header: string;
+            homepage: string | null;
+            icon: string | null;
+            icon64: string | null;
+            includes: string[];
+            lastModified: number;
+            matches: string[];
+            name: string;
+            name_i18n: Record<string, string>;
+            namespace: string | null;
+
+            /**
+             * The options that the user of the userscript
+             * can override in the settings
+             */
+            options: {
+                check_for_updates: boolean;
+                comment: string | null;
+                compat_foreach: boolean;
+                compat_metadata: boolean;
+                compat_prototypes: boolean;
+                compat_wrappedjsobject: boolean;
+                compatopts_for_requires: boolean;
+                noframes: boolean | null;
+
+                /** This uses an underscore */
+                run_at: string;
+                override: {
+                    merge_connects: boolean;
+                    merge_excludes: boolean;
+                    merge_includes: boolean;
+                    merge_matches: boolean;
+                    orig_connects: string[];
+                    orig_excludes: string[];
+                    orig_includes: string[];
+                    orig_matches: string[];
+                    orig_noframes: string | null;
+
+                    /** This also uses an underscore */
```

Comment:
What did you mean? Does property value contain underscore?

Reply:
I can remove the comments but it's just to point out the inconsistencies in the naming conventions. Two are `orig_run_at` and `run_at`, and the other is `run-at`